pkg/apis/v1/nodeclaim_conversion.go

```diff
@@ -0,0 +1,172 @@
+		return v1beta1.NodeSelectorRequirementWithMinValues{
+			NodeSelectorRequirement: v1.NodeSelectorRequirement{
+				Key:      v1Requirements.Key,
+				Operator: v1Requirements.Operator,
+				Values:   v1Requirements.Values,
+			},
+			MinValues: v1Requirements.MinValues,
+		}
+	})
+
+	if in.NodeClassRef != nil {
+		nodeclass, found := lo.Find(injection.GetNodeClasses(ctx), func(nc schema.GroupVersionKind) bool {
+			return nc.Kind == in.NodeClassRef.Kind && nc.Group == in.NodeClassRef.Group
+		})
+		v1beta1nc.NodeClassRef = &v1beta1.NodeClassReference{
+			Kind:       in.NodeClassRef.Kind,
+			Name:       in.NodeClassRef.Name,
+			APIVersion: lo.Ternary(found, nodeclass.GroupVersion().String(), ""),
+		}
+	}
+
+	// Need to implement Kubelet Conversion
+}
+
+func (in *NodeClaimStatus) convertTo(v1beta1nc *v1beta1.NodeClaimStatus) {
+	v1beta1nc.NodeName = in.NodeName
+	v1beta1nc.ProviderID = in.ProviderID
+	v1beta1nc.ImageID = in.ImageID
+	v1beta1nc.Capacity = in.Capacity
+	v1beta1nc.Allocatable = in.Allocatable
+	v1beta1nc.Conditions = in.Conditions
+}
+
+func (in *NodeClaim) ConvertFrom(ctx context.Context, from apis.Convertible) error {
+	v1beta1NC := from.(*v1beta1.NodeClaim)
+	in.Name = v1beta1NC.Name
+	in.UID = v1beta1NC.UID
+	in.Labels = v1beta1NC.Labels
+	in.Annotations = v1beta1NC.Annotations
+
+	in.Spec.convertFrom(ctx, &v1beta1NC.Spec)
+	// Updating the drift hash on the nodeclaim
+	v1beta1OwnerNodePool, err := getV1Beta1NodePoolDriftHash(ctx, v1beta1NC)
+	if err != nil {
+		return err
+	}
+	in.Annotations = lo.Assign(in.Annotations, map[string]string{
+		NodePoolHashAnnotationKey:        v1beta1OwnerNodePool.Annotations[v1beta1.NodePoolHashAnnotationKey],
+		NodePoolHashVersionAnnotationKey: v1beta1OwnerNodePool.Annotations[v1beta1.NodePoolHashVersionAnnotationKey],
+	})
+
+	in.Status.convertFrom((&v1beta1NC.Status))
+	return nil
+}
+
+func (in *NodeClaimSpec) convertFrom(ctx context.Context, v1beta1nc *v1beta1.NodeClaimSpec) {
+	in.Taints = v1beta1nc.Taints
+	in.StartupTaints = v1beta1nc.StartupTaints
+	in.Resources = ResourceRequirements(v1beta1nc.Resources)
+	in.Requirements = lo.Map(v1beta1nc.Requirements, func(v1beta1Requirements v1beta1.NodeSelectorRequirementWithMinValues, _ int) NodeSelectorRequirementWithMinValues {
+		return NodeSelectorRequirementWithMinValues{
+			NodeSelectorRequirement: v1.NodeSelectorRequirement{
+				Key:      v1beta1Requirements.Key,
+				Operator: v1beta1Requirements.Operator,
+				Values:   v1beta1Requirements.Values,
+			},
+			MinValues: v1beta1Requirements.MinValues,
+		}
+	})
+
+	nodeclasses := injection.GetNodeClasses(ctx)
+	if v1beta1nc.NodeClassRef != nil {
+		in.NodeClassRef = &NodeClassReference{
+			Name:  v1beta1nc.NodeClassRef.Name,
+			Kind:  lo.Ternary(v1beta1nc.NodeClassRef.Kind == "", nodeclasses[0].Kind, v1beta1nc.NodeClassRef.Kind),
+			Group: lo.Ternary(v1beta1nc.NodeClassRef.APIVersion == "", nodeclasses[0].Group, strings.Split(v1beta1nc.NodeClassRef.APIVersion, "/")[0]),
```

Comment:
Don't we want to change it to the new api version? If i'm misunderstanding, can you show me how the nodeClassRef.APIVersion is set?

Reply:
APIVersion is set above in the `ConvertTo` https://github.com/kubernetes-sigs/karpenter/pull/1356/files#diff-b8d4e10f084dfecb98330acb7dc0b082455afea1837c74bd59ab11d29b8518adR96